Document creating a container with volumes. Update support for Mounts.

johnflavin opened the pull request:
Addresses #228.

This PR:
* Adds a section to the README about how to mount a volume in a container,
* Deprecates the `ContainerInfo.volumes()` method in favor of `ContainerInfo.mounts()`. Information about container mounts has been returned in the latter since API 1.20.
* Adds a builder-style `Bind` inner class to `HostConfig`. Instead of passing a string into `HostConfig.builder().binds("/foo:/bar")`, now you can do `HostConfig.builder().binds(Bind.from("/foo").to("/bar").build())`.
* Slightly modifies & cleans up `ContainerMount`
* Tests creating containers with volumes and inspecting them, both in the old way and the new way.

I had some more substantial changes planned for this PR, but that's because I was basing my expectations on the Remote API docs. And when it comes to creating a container with mounts vs volumes, the docs are currently wrong. See [docker/docker#21335](https://github.com/docker/docker/issues/21335) and the comments on [docker/docker#15908](https://github.com/docker/docker/issues/15908).

mattnworb commented on src/main/java/com/spotify/docker/client/messages/ContainerMount.java:
```diff
@@ -60,29 +61,15 @@ public boolean equals(final Object o) {
-    result = 31 * result + (mode != null ? mode.hashCode() : 0);
-    result = 31 * result + (rwInfo != null ? rwInfo.hashCode() : 0);
-    return result;
+    return Objects.hashCode(source, destination, mode, rw);
```
this can just be `java.util.Objects.hash(..)` (we should replace the usage in the rest of the codebase later too)

mattnworb commented:
:+1: and thanks for the docs improvement.

Looks like a rebase is necessary, I added a similar method for testing docker versions in DefaultDockerClientTest as 

```java
// in setUp
sut = builder.build();

boolean dockerApiVersionSupported(String expected) throws Exception {
  return compareVersion(dockerApiVersion, expected) >= 0;
}
```

johnflavin commented:
In fact, I had already rebased, but I changed the name of your function from `dockerApiVersionSupported` to `dockerApiVersionGreaterOrEqualTo`. :grin:

Since it seems to have been confusing, and wasn't really necessary, I changed it back.

mattnworb commented:
ah sorry about the confusion with the method name! 

Thanks again for the contribution
